### PR TITLE
add tutorial, notebook and API doc for LightningLite

### PR DESCRIPTION
**Pull request by MeouSker77**
### 1. Why the change?

We have added `LightningLite` support in #5074 and implemented our API, we should add documentation for it.

### 2. Summary of the change

- Add API doc for `bigdl.nano.pytorch.TorchNano`
- Add a notebook for it
- Add a tutorial for it

### 3. How to test?
- [ ] N/A
- [ ] Unit test
- [ ] Application test
- [x] Document test
- [ ] ...

### CODE REVIEW

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_lightninglite.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                
+                total_loss += l.sum()
+                num += 1
+            print(f'avg_loss: {total_loss / num}')
+```
+
+### Step 5: **Run with Nano PyTorch LightningLite**
+
+```python
+Lite().run()
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_lightninglite.md:124]: change the interface to `nano().train()`? @yangw1234

File: python/nano/notebooks/pytorch/tutorial/pytorch_lightninglite.ipynb [hunk 2]
```diff
@@ -0,0 +1,361 @@
+{
```
Comment (jason-dai) [inline python/nano/notebooks/pytorch/tutorial/pytorch_lightninglite.ipynb:1]: the tutorial should be about how a PyTorch program with custom train loop can be easily accelerated by nano, not about lightning-lite

Comment (MeouSker77) [reply python/nano/notebooks/pytorch/tutorial/pytorch_lightninglite.ipynb:1]: changed it

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_lightninglite.md (hunk 1 above)
Comment (yangw1234) [reply docs/readthedocs/source/doc/Nano/QuickStart/pytorch_lightninglite.md:151]: I am OK with `nano().train()`. But is it easier for users to use if we use the same name as `PyTorchLightning`?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_nano.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        num += 1
+    print(f'avg_loss: {total_loss / num}')
+```
+
+The `TorchNano` (`bigdl.nano.pytorch.TorchNano`) class is what we use to accelerate raw pytorch code. By using it, we only need to make very few changes to accelerate custom training loop.
+
+We only need the following steps:
+
+- define a class `MyNano` derived from our `TorchNano`
+- copy all codes into the `train` method of `MyNano`
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_nano.md:111]: "code" seems to be a uncountable noun. "all lines of code"?

File: python/nano/notebooks/pytorch/tutorial/pytorch_nano.ipynb [hunk 4]
```diff
@@ -0,0 +1,463 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `TorchNano` (`bigdl.nano.pytorch.TorchNano`) class is what we use to accelerate raw pytorch code. By using it, we only need to make very few changes to accelerate custom training loop.\n",
+    "\n",
+    "We only need the following steps:\n",
+    "\n",
+    "- define a class `MyNano` derived from our `TorchNano`\n",
+    "- copy all codes into the `train` method of `MyNano`\n",
```
Comment (yangw1234) [inline python/nano/notebooks/pytorch/tutorial/pytorch_nano.ipynb:166]: "all lines of code"?

**Comment by yangw1234** [comment]
LGTM except one minor grammar issue

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_nano.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+        model, optimizer, train_loader = self.setup(model, optimizer, train_loader)
+
+        model.train()
+
+        for _i in range(max_epochs):
+            total_loss, num = 0, 0
+            for X, y in train_loader:
+                optimizer.zero_grad()
+                loss = loss_func(model(X), y)
+                self.backward(loss)     # modify this line
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_nano.md:144]: `self.backward` is still needed?

Comment (MeouSker77) [reply docs/readthedocs/source/doc/Nano/QuickStart/pytorch_nano.md:144]: yeah, if user only use `spawn`, `subprocess` and `ray` backend, then `self.backward` is not needed, but considering we may add more strategies in the future, the new strategies may need it, so we still added it

Comment (jason-dai) [reply docs/readthedocs/source/doc/Nano/QuickStart/pytorch_nano.md:144]: > yeah, if user only use `spawn`, `subprocess` and `ray` backend, then `self.backward` is not needed, but considering we may add more strategies in the future, the new strategies may need it, so we still added it

We need to figure out a way to automatically check and enforce it if we need to ask the users to call these methods.

Comment (MeouSker77) [reply docs/readthedocs/source/doc/Nano/QuickStart/pytorch_nano.md:144]: ok, I'll think of it
